### PR TITLE
[P2022-186] Three Dots Menu component

### PR DESCRIPTION
**Pull request by jprymak**

### CODE REVIEW

**Review by KarolCiesluk — APPROVED** [review]
Jakub, great job! 👍 
- it looks like  "@mui/icons-material" needs to be added to package.json

I approve but please have a look at my suggestion. Maybe you will find them relevant.

File: src/components/ThreeDotsMenu/ThreeDotsMenu.style.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { styled } from "@mui/system";
+import Menu from "@mui/material/Menu";
+
+export const StyledMenu = styled(Menu)(({ theme }) => ({
+  "& .MuiPaper-root": {
+    padding: "10px 8px",
+    color: theme.palette.grey["600"],
+    borderRadius: 4,
```
[the remark this answers is not in the document]
Comment (jprymak) [reply src/components/ThreeDotsMenu/ThreeDotsMenu.style.tsx:8]: Unfortunately it didn't work. I had copied your code from above and put it in the file just as it was, and biggerBorderRadius property couldn't be found. I will try to look into it more on the weekend.

So for now I am leaving border radius just like it was, but thank you for the idea :)

File: src/components/ThreeDotsMenu/ThreeDotsMenu.style.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    boxShadow: "0px 1px 2px rgb(98 98 98 / 0.24)",
```
[the remark this answers is not in the document]
Comment (jprymak) [reply src/components/ThreeDotsMenu/ThreeDotsMenu.style.tsx:9]: I couldn't overwrite default MuiPaper boxShadow with this sx={{ boxShadow: 1 }}. Will look into it later

File: src/components/ThreeDotsMenu/ThreeDotsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  menuItems = mockMenuItems,
+}: ThreeDotsMenuProps) {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
```
[the remark this answers is not in the document]
Comment (jprymak) [reply src/components/ThreeDotsMenu/ThreeDotsMenu.tsx]: Agreed. Done :)

File: src/components/ThreeDotsMenu/ThreeDotsMenu.style.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (KarolCiesluk) [reply src/components/ThreeDotsMenu/ThreeDotsMenu.style.tsx:8]: You are right. It doesn't work. Sorry about it, I am not so familiar with theming in mui yet. I've just written new version of theme that seems to work:
```
import { createTheme } from "@mui/material/styles";
import { Shadows } from "@mui/material/styles/shadows";

declare module "@mui/material/styles" {
  interface TypographyVariants {
    capitalHeader: React.CSSProperties;
  }

  interface TypographyVariantsOptions {
    capitalHeader?: React.CSSProperties;
  }
}

declare module "@mui/material/Typography" {
  interface TypographyPropsVariantOverrides {
    capitalHeader: true;
  }
}

declare module "@mui/system/createTheme/shape" {
  export interface Shape {
    biggerBorderRadius: number | string;
  }
}

const boxShadows = [
  "none",
  "0px 1px 3px rgba(98, 98, 98, 0.24)",
  "0px 8px 16px rgba(98, 98, 98, 0.24)",
];

export const theme = createTheme({
  palette: {
    primary: {
      main: "#fff",
    },
    secondary: {
      main: "#0d0d0d",
    },
    grey: {
      "50": "#F7F7F7",
      "100": "#E1E1E1",
      "200": "#CFCFCF",
      "300": "#B1B1B1",
      "400": "#9E9E9E",
      "500": "#7E7E7E",
      "600": "#626262",
      "700": "#515151",
      "800": "#3B3B3B",
      "900": "#222222",
    },
  },
  typography: {
    capitalHeader: {
      fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
      fontWeight: 600,
      fontSize: "12px",
      lineHeight: "24px",
      textTransform: "uppercase",
    },
  },
  shape: {
    biggerBorderRadius: 8,
  },
  shadows: boxShadows as Shadows,
});
```
I will double check it tomorrow. Hopefully it's gonna work 😄

**Review by KarolCiesluk — APPROVED** [review]

**Pull request merged**